doozerlib/cli/get_nightlies.py

```diff
@@ -1,97 +1,186 @@
+from doozerlib.model import Model
+from doozerlib import constants, util, exectools, logutil
+from doozerlib.rhcos import RHCOSBuildInspector
 
+logger = logutil.getLogger(__name__)
 
-@cli.command("get-nightlies", short_help="Get sets of Accepted nightlies. A set contains nightly for each arch, "
-                                         "determined by closest timestamps")
-@click.option("--limit", default=1, metavar='NUM', help="Number of sets of passing nightlies to print")
+
+@cli.command("get-nightlies", short_help="Determine set(s) of accepted nightlies with matching contents for all architectures")
+@click.option("--matching", metavar="NIGHTLY_NAME", multiple=True, help="Only report nightlies with the same content as named nightly")
+@click.option("--allow-pending", is_flag=True, help="Include nightlies that have not completed tests")
+@click.option("--allow-rejected", is_flag=True, help="Include nightlies that have failed tests")
+@click.option("--exclude-arch", metavar="ARCH", multiple=True, help="Exclude arch(es) normally included in this version (multi,aarch64,...)")
+@click.option("--limit", default=1, metavar='NUM', help="Number of sets of nightlies to print")
 @click.option("--rhcos", is_flag=True, help="Print rhcos build id with nightlies")
-@click.option("--latest", is_flag=True, help="Just get the latest nightlies for all arches (Accepted or not)")
+@click.option("--latest", is_flag=True, help="Just get the latest nightlies for all arches (accepted or not)")
 @click.pass_obj
-def get_nightlies(runtime, limit, rhcos, latest):
-    if latest and limit > 1:
-        print("Don't use --latest and --limit > 1", file=sys.stderr)
-        exit(1)
+@click_coroutine
+async def get_nightlies(runtime, matching: List[str], exclude_arch: List[str], allow_pending: bool, allow_rejected: bool, limit: str, rhcos: bool, latest: bool):
+    """
+    Find one or more sets of nightlies (including all arches) that have
+    matching content according to source commits and NVRs (or in the case of
+    RHCOS containers, RPM content).
+
+    By default:
+
+    * only one set of nightlies (the most recent) is displayed (see --limit)
+
+    * only accepted nightlies will be examined (see --allow-pending / --allow-rejected)
+
+    * all arches configured for the group will be included (see --exclude-arch)
+
+    You may also specify a desired nightly or nightlies (see --matching), in
+    which case those will be the only nightlies examined in their respective
+    arches.
+
+    If results do not include a nightly that you expect to see, check the
+    doozer debug.log where details about equivalence failures are logged.
+    Matching is performed in two phases:
+
+    * The first uses only info from the nightly release images for quick
+      comparison in order to construct candidate equivalent sets of nightlies.
 
+    * The second retrieves image info for all payload content in order to
+      compare group image NVRs and RHCOS RPM content.
+    """
```

Comment:
I would add some example invocations here, the normal usecase and the --matching one would help a new user

Reply:
thanks for adding this